packages/data-point/lib/trace/trace-resolve.js

```diff
@@ -0,0 +1,72 @@
+const uniqueId = require('lodash/uniqueId')
+
+const NS_PER_SEC = 1e9
+
+/**
+ * @param {hrtime} time hrtime value
+ * @returns {Integer} nanosecond
+ */
+function hrtimeTotNanosec (time) {
+  return time[0] * NS_PER_SEC + time[1]
+}
+
+module.exports.hrtimeTotNanosec = hrtimeTotNanosec
+
+/**
+ * @param {hrtime} time hrtime value
+ * @returns {Integer} duration between time argument and current time
+ */
+function getDuration (time) {
+  const diff = process.hrtime(time)
+  const nanosec = hrtimeTotNanosec(diff)
+  return nanosec
+}
+
+module.exports.getDuration = getDuration
+
+/**
+ * NOTE: Mutates traceNode argument
+ * @param {Object} traceNode
+ * @returns {Function} returns (acc):acc
+ */
+function augmentTraceNodeDuration (traceNode) {
+  return acc => {
+    traceNode.duration = module.exports.getDuration(traceNode.hrtime)
+    return acc
+  }
+}
+
+module.exports.augmentTraceNodeDuration = augmentTraceNodeDuration
+
+/**
+ * @param {Accumulator} accumulator current accumulator
+ * @param {Reducer} reducer
+ * @returns {Accumulator} traced accumulator
+ */
+function createTracedAccumulator (accumulator, reducer) {
+  const hrtime = process.hrtime()
+
+  const traceNode = {
+    id: uniqueId(),
+    reducer,
+    hrtime,
+    timeStart: hrtimeTotNanosec(hrtime),
+    parent: accumulator.traceNode
+  }
```

Comment:
to keep the trace code minimal, this could possibly be simplified:
``js
const traceNode = {
    reducer,
    hrtime,
    parent: accumulator.traceNode
  }
```
the other values could then be generated when creating the output file